src/dtemplate.d

```diff
@@ -3433,28 +3433,48 @@ extern (C++) MATCH deduceType(RootObject o, Scope* sc, Type tparam, TemplatePara
         override void visit(TypeFunction t)
         {
             //printf("TypeFunction::deduceType()\n");
-            //printf("\tthis   = %d, ", t->ty); t->print();
-            //printf("\ttparam = %d, ", tparam->ty); tparam->print();
+            //printf("\tthis   = %d, ", t.ty); t.print();
+            //printf("\ttparam = %d, ", tparam.ty); tparam.print();
+
             // Extra check that function characteristics must match
             if (tparam && tparam.ty == Tfunction)
             {
-                TypeFunction tp = cast(TypeFunction)tparam;
-                if (t.varargs != tp.varargs || t.linkage != tp.linkage)
+                auto tp = cast(TypeFunction)tparam;
+                if (t.varargs != tp.varargs ||
+                    t.linkage != tp.linkage)
                 {
                     result = MATCHnomatch;
                     return;
                 }
+
                 size_t nfargs = Parameter.dim(t.parameters);
                 size_t nfparams = Parameter.dim(tp.parameters);
+
                 // bug 2579 fix: Apply function parameter storage classes to parameter types
                 for (size_t i = 0; i < nfparams; i++)
                 {
-                    Parameter fparam = Parameter.getNth(tp.parameters, i);
+                    auto fparam = Parameter.getNth(tp.parameters, i);
                     fparam.type = fparam.type.addStorageClass(fparam.storageClass);
                     fparam.storageClass &= ~(STC_TYPECTOR | STCin);
+
+                    // Bugzilla 15243
+                    if (!reliesOnTident(fparam.type, parameters, inferStart))
+                    {
+                        Type tx = fparam.type.semantic(Loc(), sc);
+                        if (tx.ty == Terror)
+                        {
+                            result = MATCHnomatch;
+                            return;
+                        }
+                        fparam.type = tx;
```

Comment:
Thanks. From your point out, I found and got more thoughts:
In here, the original `fparam.type` never contains `Ttuple`, so we don't have to use `Parameter.dim(tp.parameters)` to iterate `tp.parameters`. Therefore we can calculate `nfparams` only once after this `for` loop.

I updated the code based on my thought.